src/controllers/user.js

```diff
@@ -34,7 +34,21 @@ const getSingleUser = async (req, res) => {
   }
 };
 
+const deleteProfile = async (req, res) => {
+  try {
+    if (req.user) {
+      await UserModel.findByIdAndDelete(req.user.userId);
+      await res.clearCookie('token');
+      return res.status(200).json({ message: 'User deleted' });
```

Comment:
you should also add a feature here to delete the products that are not given or taken from user schema.

Reply:
You mean the `isTransactionCompleted`? @okurezgi has a good point.